pkg/registry/cluster/strategy.go

```diff
@@ -84,10 +89,17 @@ func (Strategy) PrepareForCreate(ctx context.Context, obj runtime.Object) {
 
 // PrepareForUpdate is invoked on update before validation to normalize the object.
 func (Strategy) PrepareForUpdate(ctx context.Context, obj, old runtime.Object) {
-	cluster := obj.(*clusterapis.Cluster)
+	oldCluster := old.(*clusterapis.Cluster)
+	newCluster := obj.(*clusterapis.Cluster)
+
+	// Any changes to the spec increases the generation number.
+	if !apiequality.Semantic.DeepEqual(newCluster.Spec, oldCluster.Spec) {
+		newCluster.Generation = oldCluster.Generation + 1
```

Comment:
If the label changes, will it trigger rescheduling?

Reply:
As I explained, we need to discuss this. Same as `annotation`.

Reply:
https://github.com/karmada-io/karmada/blob/master/pkg/scheduler/event_handler.go#L170

Do we need to change it here as well? We should use `Generation` to determine this.

Reply:
Thanks for the reminder, and as @jwcesign said, combined with the usage in the scheduler event handler, I think we can take the label changes into account.

Reply:
Seems there is no guideline about if a `generation` should be increased in case of `label` updation, but `no resource in k8s will do that`.

The only benefit we can get from that is the scheduler doesn't have to compare the labels here:
https://github.com/karmada-io/karmada/blob/2c834fe94eb2eff8a9c6954a4c2b0ad567d6caf1/pkg/scheduler/event_handler.go#L168-L172

Please remind me if I miss any benefits. 
It's hard to make the choice, I'd like your opinion. @XiShanYongYe-Chang @chaunceyjiang @Garrybest

Reply:
It's really hard to choose. From an unknown future expansion point of view, addition is better than subtraction, so I think it might be better not to consider labels here.

Reply:
I prefer to compare only the spec section here, and then use `generation` and `label` in the event_handler.go.

Reply:
Thanks, then please @XiShanYongYe-Chang make changes.